srcalc/src/test/java/gov/va/med/srcalc/vista/RpcVistaPatientDaoTest.java

```diff
@@ -370,4 +386,44 @@ public final void testInvalidDnrNotes() throws Exception
         // This would fail to retrieve DNR notes and the notes would be empty.
         assertEquals(0, patient.getDnrNotes().size());
     }
+    
+    @Test
+    public final void testFullVitals() throws Exception
+    {
+        final VistaProcedureCaller caller = mockVistaProcedureCaller();
+        when(caller.doRpc(
+                RADIOLOGIST_DUZ, RemoteProcedure.GMV_LATEST_VM, String.valueOf(PATIENT_DFN)))
+            .thenReturn(FULL_VITALS);
+        final RpcVistaPatientDao dao = new RpcVistaPatientDao(caller, RADIOLOGIST_DUZ);
+        final Patient patient = dao.getPatient(PATIENT_DFN);
+        final ImmutableList<Double> expectedValues = ImmutableList.of(178.0, 71.0, 24.88);
+        assertEquals(expectedValues, ImmutableList.of(patient.getWeight().getValue(),
+                patient.getHeight().getValue(), patient.getBmi().getValue()));
+    }
+    
+    @Test
+    public final void testPartialVitals() throws Exception
+    {
+        final VistaProcedureCaller caller = mockVistaProcedureCaller();
+        when(caller.doRpc(
+                RADIOLOGIST_DUZ, RemoteProcedure.GMV_LATEST_VM, String.valueOf(PATIENT_DFN)))
+            .thenReturn(PARTIAL_VITALS);
+        final RpcVistaPatientDao dao = new RpcVistaPatientDao(caller, RADIOLOGIST_DUZ);
+        final Patient patient = dao.getPatient(PATIENT_DFN);
+        final ImmutableList<Double> expectedValues = ImmutableList.of(72.0, 208.0, 28.27);
+        assertEquals(expectedValues, ImmutableList.of(patient.getHeight().getValue(),
+                patient.getWeight().getValue(), patient.getBmi().getValue()));
+    }
+    
+    @Test
+    public final void testEmptyVitals()
+    {
+        final VistaProcedureCaller caller = mockVistaProcedureCaller();
+        final RpcVistaPatientDao dao = new RpcVistaPatientDao(caller, RADIOLOGIST_DUZ);
+        final Patient patient = dao.getPatient(PATIENT_DFN);
+        // Empty vitals means the vitals do not get filled with any values.
+        assertEquals(null, patient.getWeight());
+        assertEquals(null, patient.getHeight());
+        assertEquals(null, patient.getBmi());
+    }
```

Comment:
You should also add a test for unexpected data coming from VistA. Make sure your code is robust.